python/lancedb/embeddings/base.py

```diff
@@ -125,6 +137,12 @@ def VectorField(self, **kwargs):
         """
         return Field(json_schema_extra={"vector_column_for": self}, **kwargs)
 
+    def __eq__(self, __value: object) -> bool:
+        return vars(self) == vars(__value)
```

Comment:
if the input does not have a __dict__ attr then this raises a TypeError. e.g., try `vars("foo")`

Reply:
Yeah it should fail with any object that does not have a `__dict__` attr. All class instances should have a dict attribute. This is like overriding equlity operator so it's only supposed to be called with an object of same type.

```
class A

a = A()
b = A()
a == b
```
Used by hash function which is in turn used for creating weak references for caching